diff/diff.go

```diff
@@ -0,0 +1,276 @@
+package diff
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/fatih/color"
+)
+
+// Opt is an option type that can be passed to New.
+type Opt func(Differ) Differ
+
+// WithFormat returns an Opt that wraps up differences
+// using a format string.  The format should contain
+// one '%s' to add the difference string in.
+func WithFormat(format string) Opt {
+	return func(d Differ) Differ {
+		d.wrappers = append(d.wrappers, func(v string) string {
+			return fmt.Sprintf(format, v)
+		})
+		return d
+	}
+}
+
+// Style represents display styles (like bold or italic)
+// that we can display text as.
+type Style int
+
+const (
+	Bold Style = 1 + iota
+	Faint
+	Italic
+	Underline
+	CrossedOut
+)
+
+// WithStyle returns an Opt that wraps up differences
+// in a style.
+func WithStyle(s Style) Opt {
+	switch s {
+	case CrossedOut:
+		// We aren't matched 1:1 with fatih/color on this attribute.
```

Comment:
Because I didn't want to include stuff like `Blink`.  `github.com/fatih/color` is currently unmaintained, so I didn't want to support everything that it can do and lock ourselves into only using that library.  If we find a better one, I don't want us to have a hard requirement that they support blinking text and junk.

Reply:
While it's technically unmaintained, it's still the standard library for this type of thing. I'm not sure it makes sense to expand onpar's API to include color stuff